container_pipeline/utils.py

```diff
@@ -20,6 +22,45 @@ def get_project_name(job):
     return '{}-{}-{}'.format(job['appid'], job['jobid'], job['desired_tag'])
 
 
+def parse_json_response(response):
+    """Parses the json response provided to it to determin"""
+    try:
+        cause = response['actions'][0]['causes'][0]['shortDescription']
+
+        if cause == 'Started by an SCM change':
+            return 'Git commit {}'.format(
+                response['actions'][2]['lastBuiltRevision']['SHA1']
+            )
+        elif 'Started by upstream project' in cause:
+            return 'Change in upsream project {}'.format(
+                response['actions'][0]['causes'][0]['shortDescription'].split(
+                    '"')[1]
+            )
+        elif 'Started from command line' in cause:
+            return 'RPM update in enabled repos'
+        elif 'Started by user' in cause:
+            return 'Manually triggered by admin'
+        else:
+            return 'Unknown'
+    except KeyError:
+        return 'Invalid JSON response from Jenkins'
+
+
+def get_cause_of_build(jenkins_url, job_name, job_number):
+    """
+    This function expects Jenkins endpoint, job name and job number to figure
+    the cause of build
+    """
+    try:
+        url = "http://{}:8080/job/{}/{}/api/json".format(
+            jenkins_url, job_name, job_number)
+        response = urllib2.urlopen(url)
+    except URLError:
+        return "Invalid URL: {}".format(url)
+    else:
+        return parse_json_response(json.loads(response.read()))
```

Comment:
may be we can keep `parse_json_respnse ` inside of try. so that we can the parse error also if there is any.

Reply:
We have a `try..except` block in `parse_json_response` to handle its only possible `KeyError`. So I kept that in `else` block.